### PR TITLE
Add no-var and prefer-const rules

### PR DESCRIPTION
**Pull request by aaccurso**
This configuration will throw error upon usage of `var` to declare variables, helping to clarify the scope of a variable.

Also it will throw error when using `let` to declare variables that aren't reassigned post declaration, promoting the use of `const`.

Some interesting related articles:

- https://ponyfoo.com/articles/var-let-const
- https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
Thoughts on setting [`"destructuring": "all"`](http://eslint.org/docs/rules/prefer-const#options)? Otherwise, def 👍.

**Comment by aaccurso** [comment]
Didn't know that one, I vote 👍 . It will definitely sharpen our React's `render` methods.
However, I'd include that in a different PR. I'd like to test some configurations of it on Samsung's KC if that's ok.

File: index.js
Comment (mAAdhaTTah) [inline index.js]: Add `"destructuring": "all"` for `prefer-const` and you've got my approval.

**Review by mAAdhaTTah — CHANGES_REQUESTED** [review]

**Comment by MichaelFoss** [comment]
@aaccurso, I will approve this once the `"destructuring": "all"` has been added.

**Comment by aaccurso** [comment]
Just so we are on the same page, are we talking about http://eslint.org/docs/rules/prefer-destructuring , right? Because that rule is called `prefer-destructuring`.

Or is it this plugin https://www.npmjs.com/package/eslint-plugin-destructuring ?

Cause I've used neither on a project so I think it'd be irresponsible on my part if I created a PR with something I haven't tried 😃

**Comment by mAAdhaTTah** [comment]
@aaccurso See the link provided by my comment. It's an option for `prefer-const`.

**Review by mAAdhaTTah — APPROVED** [review]

**Review by MichaelFoss — APPROVED** [review]

**Comment by mAAdhaTTah** [comment]
@MichaelFoss + me = 2 approvals. Merging this PR.

**Pull request merged**
